apache/trafficcontrol
# Make `db/admin` use Migrate to migrate the Traffic Ops database

Closes: #3342

This PR makes the Traffic Ops `db/admin` tool use [Migrate](https://github.com/golang-migrate/migrate), removes Goose, and includes a migration to migrate from Goose to Migrate.

---


## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Control Cache Config (T3C) (integration tests)
- Traffic Ops (database, `db/admin` tool)
- CDN in a Box
- Automation (GitHub Actions)

## What is the best way to verify this PR?

-   Run migrations after creating the database
    ```shell
    cd traffic_ops/app
    db/admin -env production reset
    ```

-   Reset the DB, run migrations using `goose`, ensure that `db/admin` continues at the correct migration version:
    ```shell
    cd traffic_ops/app
    mv db/migrations db/migrations.migrate
    git checkout b7ced4a8c1 -- db/migrations
    git reset -- db/migrations
    goose -env production up
    mv db/migrations db/migrations.goose
    mv db/migrations.migrate db/migrations
    db/admin -env production migrate
    db/admin -env production dbversion
    ```
    
    Expected output of last 2 commands (`db/admin migrate and dbversion`):
    ```    
    0/u init (170.821438ms)
    dbversion 2021070800000000
    ```

-   Create an up/down migration and verify that the created migration has the correct number of digits
    ```shell
    db/admin create_migration my-migration
    ```

-   Run tests

## PR submission checklist
- [x] This PR updates existing tests
